spring-cloud-app-broker-logging/src/main/java/org/springframework/cloud/appbroker/logging/streaming/endpoint/StreamingLogWebSocketHandler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.logging.streaming.endpoint;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.cloudfoundry.dropsonde.events.Envelope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+
+import org.springframework.cloud.appbroker.logging.streaming.events.ServiceInstanceLogEvent;
+import org.springframework.cloud.appbroker.logging.streaming.events.StartServiceInstanceLoggingEvent;
+import org.springframework.cloud.appbroker.logging.streaming.events.StopServiceInstanceLoggingEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationListener;
+import org.springframework.web.reactive.socket.WebSocketHandler;
+import org.springframework.web.reactive.socket.WebSocketSession;
+import org.springframework.web.util.UriTemplate;
+
+public class StreamingLogWebSocketHandler implements WebSocketHandler, ApplicationListener<ServiceInstanceLogEvent> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(StreamingLogWebSocketHandler.class);
+
+	private static final UriTemplate LOGGING_URI_TEMPLATE = new UriTemplate("/logs/{serviceInstanceId}/stream");
+
+	private final ApplicationEventPublisher eventPublisher;
+
+	private final ConcurrentHashMap<String, Sinks.Many<Envelope>> envelopeSinks = new ConcurrentHashMap<>();
```

Comment:
```suggestion
	private final ConcurrentMap<String, Sinks.Many<Envelope>> envelopeSinks = new ConcurrentHashMap<>();
```

Reply:
Applying it in a different commit to add the import. Thanks!